src/writable.ts

```diff
@@ -99,7 +97,11 @@ class WritableBundle<T> {
 				throw new Error('This bundle has already been finalized');
 			}
 
-			const name = `${RESOURCES_DIR}/` + hasher.checksum;
+			if (this.addedResources.has(id)) {
+				throw new Error(`Resource "${id}" is already added`);
+			}
```

Comment:
I think this isn't necessary because a resource can only be added if it has been declared in the `resources` array, and we check for duplicates in it early on.

Reply:
I thought initially so as well, but realized one could add the same resource twice although there are no duplicates in the `resources` array. I added `add resource which was already added` test that expects this new exception.